File: android/app/src/main/java/com/zillit/lcw/data/model/ActivityLog.kt
package com.zillit.lcw.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ActivityLog(
    @SerialName("_id") val id: String,
    val projectId: String,
    val action: String, // created, updated, deleted, duplicated, shared
    val targetType: String, // schedule_day, schedule_type, event, note
    val targetId: String = "",
    val targetTitle: String = "",
    val details: String = "",
    val performedBy: PerformedBy = PerformedBy(),
    val createdAt: Long = 0L,
    val updatedAt: Long = 0L
)

@Serializable
data class PerformedBy(
    val userId: String = "",
    val name: String = ""
)

File: ios/ZillitLCW/Models/ActivityLog.swift
import Foundation

struct ActivityLog: Identifiable, Codable {
    let id: String
    let projectId: String
    let action: String // created, updated, deleted, duplicated, shared
    let targetType: String // schedule_day, schedule_type, event, note
    let targetId: String
    let targetTitle: String
    let details: String
    let performedBy: PerformedBy
    let createdAt: Int64

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case projectId, action, targetType, targetId, targetTitle, details, performedBy, createdAt
    }
}

struct PerformedBy: Codable {
    let userId: String
    // `name` is no longer returned by the API — resolved client-side via UserCache.
    // Kept as an optional for compatibility with any locally-cached older payloads.
    let name: String?

    init(userId: String, name: String? = nil) {
        self.userId = userId
        self.name = name
    }
}
